blueprints/hooks/light/light.yaml

```
@@ -298,19 +307,27 @@ action:
               transition: '{{ light_transition / 1000 }}'
       - conditions: '{{ action == brightness_up }}'
         sequence:
-          - service: light.turn_on
-            data:
-              brightness: '{{ min_brightness if smooth_power_on and states(light) == "off" else [ [state_attr(light,"brightness")+step_short, min_brightness] | max, max_brightness] | min }}'
-              transition: 0.25
-              entity_id: !input light
-      - conditions: '{{ action == brightness_down }}'
+          - choose:
+              - conditions: '{{ smooth_power_on == false and states(light) == "off" and state_attr(light,"brightness") == none }}'
+                sequence:
+                  - service: light.turn_on
+                    data:
+                      transition: '{{ light_transition / 1000 }}'
+                      entity_id: !input light
+            default:
+              - service: light.turn_on
+                data:
+                  brightness: '{{ min_brightness if smooth_power_on and states(light) == "off" else [ [state_attr(light,"brightness")+step_short, min_brightness] | max, max_brightness] | min }}'
+                  transition: 0.25
+                  entity_id: !input light
+      - conditions: '{{ action == brightness_down and states(light) == "on" }}'
         sequence:
           - choose:
               - conditions: '{{ smooth_power_off and state_attr(light,"brightness") == min_brightness }}'
                 sequence:
                   - service: light.turn_off
                     data:
-                      transition: 0.25
+                      transition: '{{ light_transition / 1000 }}'
```

```suggestion
                      transition: 0.25
```

Watch out here, the `light_transition` input is currently used only to control transition duration when turning on and off the light. The PR #33 adds the ability to configure the transition time of each step, when controlling brightness and color temperature. That PR was waiting for feedback from the original contributor, but since it has been stayed still without any response for more than 4 months now, I think it's time to merge it. :)

---

pretty sure thats how i implemented it - could you take anothere look?

Should only be on lines where the light is either being turned on or off

---

You're right! Sorry for my wrong report. :)